### PR TITLE
Refactor eliot-tree into smaller pieces, with tests, that can be used by third parties

### LINKED ISSUE #16: Factor out core functionality into a library
Currently, eliottree is a single script that does a bunch of stuff. If the core functionality were factored out into a library that the script used, it would be more maintainable (and testable!) and allow other tools to more easily build on top of eliottree.

### PR DESCRIPTION
**Pull request by jonathanj**
Closes #16.

### CODE REVIEW

**Comment by jerith** [comment]
Looks reasonable to me. I haven't done a thorough review, because I'm not really familiar with the code, but I don't see any glaring issues.

Could we perhaps hook this up to travis-ci.org or some other continuous integration system?

**Comment by jerith** [comment]
Tests for `eliottree._cli` would be nice, but I don't think that necessarily has to block this PR. Other than that, :+1: from me.

Also, you can configure coveralls to not comment on pull requests (I find that a bit noisy, but it's your call) and you can set the allowed reduction in coverage before coveralls returns a build failure status.

**Comment by jonathanj** [comment]
Thanks @jerith, I've configured coveralls to be less spammy, I didn't know you could configure that stuff.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+[testenv:py27]
+basepython = python2.7
+deps =
+    {[testenv]deps}
+
+[testenv:pep8]
+deps = pep8
+basepython = python2.7
+commands = pep8 eliottree
```
Comment (mithrandi) [inline tox.ini]: There should be a newline at the end of the file.

**Comment by mithrandi** [comment]
:+1: from me as well, aside from the thing I noted above.

**Pull request merged**